include/simdjson/dom/parser-inl.h

```diff
@@ -168,17 +175,31 @@ inline bool parser::allocate_capacity(size_t capacity, size_t max_depth) noexcep
   return !allocate(capacity, max_depth);
 }
 #endif // SIMDJSON_DISABLE_DEPRECATED_API
-inline error_code parser::ensure_capacity(size_t desired_capacity) noexcept {
+
+inline error_code parser::ensure_capacity_with_document(size_t desired_capacity) noexcept {
```

Comment:
This does the opposite of what I guessed from the name when it was called (it sounds like something where I get to pass a different document rather than ensuring the capacity of the parser itself). Minor note, don't really ultimately care, I just sometimes point out what I see so it doesn't pass unsaid.

Reply:
I have reverted this to the old name. The reason I had created this ugly function (that matches the old ensure_capacity) is that I had changed the behaviour of ensure_capacity, but it turns out that there is never a case where you'd want to "ensure_capacity" without also touching a document. So I have now reverted the name to the old name (which I think you came up with so you should approve).